python/lsst/afw/table/aggregates.cc

```diff
@@ -21,25 +21,185 @@
+    clsPointKey.def(py::self == py::self);
+    clsPointKey.def(py::self != py::self);
+    /* Members */
+    clsPointKey.def("getX", &PointKey<T>::getX);
+    clsPointKey.def("getY", &PointKey<T>::getY);
+    clsPointKey.def("isValid", &PointKey<T>::isValid);
+    clsPointKey.def_static("addFields", &PointKey<T>::addFields);
+    clsPointKey.def("set", [](PointKey<T> & pointKey, BaseRecord & record, lsst::afw::geom::Point<T,2> const & value){
+        return pointKey.set(record, value);
+    });
+    clsPointKey.def("get", &PointKey<T>::get);
+};
+
+template <typename T, int N>
+void declareCovarianceMatrixKey(py::module &mod, const::std::string & suffix){
+    typedef std::vector< Key<T> > SigmaKeyArray;
+    typedef std::vector< Key<T> > CovarianceKeyArray;
+    typedef std::vector<std::string> NameArray;
+    
+    py::class_<CovarianceMatrixKey<T,N>> cls(mod, ("CovarianceMatrix"+suffix+"Key").c_str());
+    
+    cls.def(py::init<>());
+    cls.def(py::init<SigmaKeyArray const &, CovarianceKeyArray const &>(),
+            "sigma"_a, "cov"_a=CovarianceKeyArray());
+    cls.def(py::init<SubSchema const &, NameArray const &>());
+    
+    cls.def_static("addFields", (CovarianceMatrixKey<T,N> (*)(
+            Schema &,
+            std::string const &,
+            NameArray const &,
+            std::string const &,
+            bool)) &CovarianceMatrixKey<T,N>::addFields,
+        "schema"_a, "prefix"_a, "names"_a, "unit"_a, "diagonalOnly"_a=false
+    );
+    cls.def_static("addFields", (CovarianceMatrixKey<T,N> (*)(
+            Schema &,
+            std::string const &,
+            NameArray const &,
+            NameArray const &,
+            bool)) &CovarianceMatrixKey<T,N>::addFields,
+        "schema"_a, "prefix"_a, "names"_a, "units"_a, "diagonalOnly"_a=false
+    );
+    cls.def("set", [](CovarianceMatrixKey<T,N> & cov, BaseRecord & record, Eigen::Matrix<T,N,N> const & value){
+        return cov.set(record, value);
+    });
+    cls.def("get", [](CovarianceMatrixKey<T,N> & cov, BaseRecord const & record){
+        return cov.get(record);
+    });
+    cls.def("isValid", &CovarianceMatrixKey<T,N>::isValid);
+    cls.def("setElement", &CovarianceMatrixKey<T,N>::setElement);
+    cls.def("getElement", &CovarianceMatrixKey<T,N>::getElement);
+    cls.def("__eq__", [](CovarianceMatrixKey<T,N> & cov, CovarianceMatrixKey<T,N> & other){
+        return cov==other;
+    });
+    cls.def("__ne__", [](CovarianceMatrixKey<T,N> & cov, CovarianceMatrixKey<T,N> & other){
+        return cov!=other;
+    });
+};
+
 PYBIND11_PLUGIN(_aggregates) {
     py::module mod("_aggregates", "Python wrapper for afw _aggregates library");
 
+    if (_import_array() < 0) {
+            PyErr_SetString(PyExc_ImportError, "numpy.core.multiarray failed to import");
+            return nullptr;
+    };
+
     /* Module level */
+    declarePointKey<double>(mod, "2D");
+    declarePointKey<int>(mod, "2I");
+    declareCovarianceMatrixKey<float,2>(mod, "2f");
```

Comment:
Jim is correct (see https://github.com/lsst/afw/blob/057223eaf06faccd9bb52423f8e0505a01a9cbc6/python/lsst/afw/table/aggregates.i)